client/js/components/main/assessment.jsx

```diff
@@ -46,6 +48,46 @@ export class Assessment extends React.Component{
   }
 
   /**
+   * Will check if assessment is completed and then dispatch submit assessment
+   * action
+   */
+  submitAssessment(){
+    var complete = this.checkCompletion();
+    if(complete === true){
+      window.onbeforeunload = null;
+      this.props.submitAssessment(
+        this.props.assessment.id,
+        this.props.assessment.assessmentId,
+        this.props.allQuestions,
+        this.props.responses,
+        this.props.settings,
+        this.props.outcomes
+      );
+    } else {
+      //TODO display some sort of warning message?
+    }
+  }
+
+  /**
+   * Returns true if all questions have been answered, false otherwise.
+   */
```

Comment:
I will implement this in my next branch where I have more of an Idea of how question responses will work.